File: components/PSU.java
package components; // ← Fará parte do pacote "components", junto com outras peças de computador.
// ↓ Criando a classe PSU (fonte de alimentação)
public class PSU {
 private String nome; // ← Nome da fonte de alimentação
 private Integer w; // ← Watts como corrente contínua
 private Integer ef; // ← Eficiência: relação entre contínua e alternada
 private Integer ac; // ← Watts como corrente alternada
 private Integer calor; // ← Calor gerado pela fonte
 private String certif; // ← Certificações 80 Plus
 // ↓ Argumentos para criar a classe
 public PSU (String nome, Integer w, Integer ef) {
  // ↓ NOTA: Os tratadores de exceção abaixo, marcados com if, são personalizados.
  if (nome == null || nome.trim().isEmpty()) {throw new IllegalArgumentException("Não pode ser vazio ou nulo!");}
  if (w <= 0) {throw new IllegalArgumentException("Deve ser maior que zero!");}
  if (ef < 1 || ef > 100) {throw new IllegalArgumentException("Deve ser entre 1 e 100!");}
  this.nome=nome;
  this.w=w;
  this.ef=ef;
  this.ac = (this.w * 100 / this.ef);
  this.calor = this.ac - this.w;
  this.certif = (ef >= 94) ? "80 Plus Titanium" :
                (ef >= 91) ? "80 Plus Platinum" :
                (ef >= 88) ? "80 Plus Gold" :
                (ef >= 85) ? "80 Plus Silver" :
                (ef >= 82) ? "80 Plus Bronze" :
                (ef >= 80) ? "80 Plus" : "Não";
 }
 // ↓ Simplesmente digitando o nome do instance dentro do parênteses no "System.out.println()" irá renderizar a string abaixo
 @Override
 public String toString() {
  return "\n::::: PROPRIEDADES DA FONTE DE ALIMENTAÇÃO :::::\nNome: "+nome+"\nCorrente contínua: "+w+"W\nCorrente alternada: "+ac+"W\nEficiência: "+ef+"%\nCalor: "+calor+"W\nCertificação: "+certif+"\n";
 }
 // ↓ NOTA: Para manter os atributos da classe como private e para serem visíveis por outras classes, os getters abaixo serão utilizados.
 public String verNome() {return nome;}
}

File: components/PSU.py
# ↓ Criando a classe PSU (fonte de alimentação)
class PSU:
 # ↓ Argumentos para criar a classe
 def __init__ (self, nome:str, w:int, ef:int):
 
  # ↓ NOTA: Os tratadores de exceção abaixo, marcados com if, são personalizados.
  if nome is None or nome.strip() == "":
   raise ValueError("Não pode ser vazio ou nulo!")
  if w <= 0:
   raise ValueError("Deve ser maior que zero!")
  if ef < 1 or ef > 100:
   raise ValueError("Deve ser entre 1 e 100!")
  
  self.__nome=nome
  self.__w=w
  self.__ef=ef
  self.__ac = self.__w * 100 / self.__ef
  self.__calor = self.__ac - self.__w
  self.__certif = "80 Plus Titanium" if self.__ef >= 94 else "80 Plus Platinum" if self.__ef >= 91 else "80 Plus Gold" if self.__ef >= 88 else "80 Plus Silver" if self.__ef >= 85 else "80 Plus Bronze" if self.__ef >= 82 else "80 Plus" if self.__ef >= 80 else "Nenhuma"
                
 # ↓ Simplesmente digitando o nome do instance no console, ou dentro do parênteses no "print()" irá renderizar a string abaixo
 def __repr__(self) -> str:
  return f"\n::::: PROPRIEDADES DA FONTE DE ALIMENTAÇÃO :::::\nNome: {self.__nome}\nCorrente contínua: {self.__w}W\nCorrente alternada: {self.__ac}W\nEficiência: {self.__ef}%\nCalor: {self.__calor}W\nCertificação: {self.__certif}\n";

 # ↓ NOTA: Para manter os atributos da classe como privados e para serem visíveis por outras classes, os getters abaixo serão utilizados.
 def verNome(self) -> str:
  return self.__nome
 def verCorrenteContinua(self) -> int:
  return self.__w
 def verCorrenteAlternada(self) -> int:
  return self.__ac
 def verEficiencia(self) -> int:
  return self.__ef
 def verCalor(self) -> int:
  return self.__calor
 def verCertificado(self) -> str: 
  return self.__certif
